requirements/specifications/lifecycle/index.md

```diff
@@ -1,274 +1,365 @@
+to interact to ensure the transition goes smoothly.
+
+![Lifecycle States](../../images/specifications/lifecycle/lifecycle-transitions.png)
+
+As an app changes states the platform will invoke specific app-provided transition
+methods from the `Application` interface:
+
+|   | Legend |
+| - | ------ | 
+| ↓ | Limited |
+| ✔ | Normal access |
+
+|                | CPU | RAM | Net | GFX | A/V | Description                                                                   |
+|----------------|-----|-----|-----|-----|-----|-------------------------------------------------------------------------------|
+| `create()`     | ✔   | ✔   | ✔   | ✔   |     | Creation of the app, from a Firebolt Lifecycle perspective.                   |
+| `suspend()`    | ✔   | ✔   | ✔   | ✔   |     | Reduce memory usage and CPU cycles, deallocate graphics compositors.          |
+| `resume()`     | ✔   | ✔   | ✔   | ✔   |     | Graphics compositors reallocated, full memory usage and normal CPU cycles.    |
+| `destroy()`    | ✔/↓ | ✔/↓ | ✔   |     |     | Preprare for the app to be deallocated and removed from execution. CPU & RAM based on previous state. |
+
+If an app implements the `Activity` interface, then the following transitions may be invoked:
 
-Finally, the app and its container **MUST** be removed from memory and
-have other resources released as well.
+|                | CPU | RAM | Net | GFX | A/V | Description                                                                   |
+|----------------|-----|-----|-----|-----|-----|-------------------------------------------------------------------------------|
+| `activate()`   | ✔   | ✔   | ✔   | ✔   | ✔   | App is expected to become a user-perceptible part of the user experience.     |
+| `deactivate()` | ✔   | ✔   | ✔   | ✔   | ✔   | Must remove any user-perceptible activities and deallocate A/V decoders.      |
 
-Apps may optionally implement the Lifecycle.ResourceDeallocator API for
-more control over app suspension.
 
-## 3. Lifecycle State Transitions
+Finally, if an app implements the `Sleepable` interface, then the following transistions may be invoked.
 
-Firebolt platforms **MUST** support the app lifecycle states and
-transitions denoted in blue.
+|                | CPU | RAM | Net | GFX | A/V | Description                                                                   |
+|----------------|-----|-----|-----|-----|-----|-------------------------------------------------------------------------------|
+| `sleep()`      | ↓   | ↓   | ✔   |     |     | Prepare for an extended period with no CPU cycles given to app.               |
+| `wake()`       | ↓   | ↓   | ✔   |     |     | Cleanup after an extended period with no CPU, e.g. reset timers / network connections.   |
 
-Platforms **MAY** support the app lifecycle states and transitions
-denoted in gray.
+These transition APIs are asynchronous, and each one has a platform-configurable timeout that specifies how long the app has to fulfill the method.
```

Comment:
Yeah, Firebolt is waiting until the APIs return,so I think that makes it sync?
But maybe we just don't mention async/sync here since it could be confusing?